imap/sync_support.c

```
@@ -3866,17 +3867,47 @@ int sync_apply_rename(struct dlist *kin, struct sync_state *sstate)
     }
 
     r = mboxlist_lookup_allow_all(oldmboxname, &mbentry, 0);
+    if (r) goto done;
+
+    if (mboxname_isusermailbox(oldmboxname, 1) &&
+        mboxname_isusermailbox(newmboxname, 1)) {
+        // we can't rename users if the new inbox already exists!
+        mbentry_t *destmbentry = NULL;
+        r = mboxlist_lookup_allow_all(newmboxname, &destmbentry, NULL);
+        if (r != IMAP_MAILBOX_NONEXISTENT) {
+            mboxlist_entry_free(&destmbentry);
+            r = IMAP_MAILBOX_EXISTS;
```

(Same concern here about hiding other errors)